### ISSUE TITLE
Impossibility parse jsonLogic with custom operator

### ISSUE THREAD

**Issue by shadowrix** [react-querybuilder/react-querybuilder#433]
I use the formatQuery function and convert to jsonLogic format with custom operators. Example:
```
export function customRulesJsonLogic(
  rule: RuleType,
  options?: ValueProcessorOptions,
) {
  switch (rule.operator) {
    case '==':
      return { '==': [rule.value, { var: rule.field }] }
    case 'regex':
      return { regex: [rule.value, { var: rule.field }] }
    default:
      return defaultRuleProcessorJsonLogic(rule, options)
  }
}
```
```
formatQuery(query, {
          format: 'jsonlogic',
          ruleProcessor: customRulesJsonLogic,
        })
```

Then I want to parse back from jsonLogic and use the parseJsonLogic function. But it doesn't work with custom operators. If the operator "==" comes across, it parses it into "=", and if "regex", it does not parse at all.
```
parseJsonLogic(jsonLogic)
```

Is it possible to add custom operators for the parser somehow?

**Comment by jakeboone02** [comment]
Hi @LaCuba! @hwhmeikle asked a similar question in #398. If that thread doesn't help or point you in the right direction, let me know and I'll see what else we can do.

**Comment by shadowrix** [comment]
@jakeboone02 Thanks for the very quick reply! 
I would really like this to solve the problem, but unfortunately we will not succeed because we need all the standard operators and add new ones to them.

**Comment by jakeboone02** [comment]
OK. How about something like this:

```ts
const query = parseJsonLogic(myJsonLogicObject, {
  jsonLogicOperations: {
    regex: (val) => ({
      field: val[0], operator: 'contains', value: val[1]
    })
  }
})
```

The `jsonLogicOperations` option would take an object where the keys would be the custom operations and the associated values would be functions that take an `any` variable (whatever is associated to the custom operation) and return a `RuleType` or even `RuleGroupType`.

**Comment by shadowrix** [comment]
It will be just wonderful!

**Comment by jakeboone02** [comment]
@LaCuba I just released [v5.3.2](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v5.3.2) with a new configuration option for `parseJsonLogic` called `jsonLogicOperations`. Documentation [here](https://react-querybuilder.js.org/docs/api/import#custom-operations).

You can now specify a function that returns a rule or group based on any custom operation.

**Comment by shadowrix** [comment]
I am very grateful to you! It was very fast! I also want to thank you for this wonderful library!

**Comment by shadowrix** [comment]
@jakeboone02 There is a non-default behavior: This example below does not work correctly, returning a value without a combinator.
```
parseJsonLogic(
  {
    'regex': [
      'XXXX',
      {
        var: 'point',
      },
    ],
  },
  {
    jsonLogicOperations: {
      'regex': (val) => ({
          field: val[1].var,
          operator: 'regex',
          value: val[0],
        })
    },
  },
)
```
Returned value:
```
{
    "field": "point",
    "operator": "regex",
    "value": "XXXX"
}
```
If you do the same with default operators, everything will work correctly. Returned value:
```
{
    "combinator": "and",
    "rules": [
        {
            "field": "point",
            "operator": "!=",
            "value": "XXXX"
        }
    ]
}
```

**Comment by jakeboone02** [comment]
Oops, good catch! [Fixed in v5.3.3](https://github.com/react-querybuilder/react-querybuilder/releases/tag/v5.3.3).

**Comment by shadowrix** [comment]
Thank you very much!
